packages/sdk/src/evm/core/classes/base-router.ts

```diff
@@ -110,7 +110,7 @@ export class BaseRouterClass<TContract extends BaseRouter>
     },
   );
 
-  addExtensionWithAbi = /* @__PURE__ */ buildTransactionFunction(
+  addWithAbi = /* @__PURE__ */ buildTransactionFunction(
```

Comment:
how can `extensionAbi` be optional here? you def need it.

also, can you add `addPublished(extensionName, publisher?)` ? that one should pull the abi and uri from publish metadata